Game/src/Collision.java

```diff
@@ -3,19 +3,20 @@
  * Date: 24/10/2018
  */
 
-public class Collision{
+public class Collision {
 
     /**
-     * Right edge of player is colliding.
+     * Player Collision Right
      * @param player The player.
      * @return The right edge of the player is colliding with any static map object (StaticRect) in sceneObjects.
      */
-    public StaticRect isCollidingRight(Player player){
+    public StaticRect isCollidingRight(Player player) {
 
-        for( Object object : MainApplication.sceneObjects){
-            if ( object instanceof StaticRect ) {
+        for (Object object : MainApplication.sceneObjects) {
+            if (object instanceof StaticRect) {
                 StaticRect staticRect = (StaticRect) object;
-                if (staticRect.checkCollision(player).contains(StaticRect.CollisionType.Right)) {
+
+                if (staticRect.checkCollision(player).contains(StaticRect.CollisionType.Right)  && staticRect.getCollisionEnabled()) {
```

Comment:
Something you might not know about in programming is something called "Short Circuit Evaluation." Basically if the first condition of an AND operator is False then the rest of the statement won't be evaluated (as it doesn't matter). 
False && True = False && False = False

The same holds true for an OR operator but for True statements instead of False ones:
True || False = True || True = True

The take away from this is that if you have a line of code that will be executed all the time, then you should put the condition that is faster to check first. This will greatly speed up your code. In this case checking if collision is enabled is much much faster than testing if it colliding with the player.